admin/homebrew.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Create a Homebrew recipe for Flocker, using the VERSION environment variable.
+
+Inspired by https://github.com/tdsmith/labmisc/blob/master/mkpydeps.
+"""
+
+from os import environ
+from json import load
+from urllib2 import HTTPError, urlopen
+from hashlib import sha1
+from tl.eggdeps.graph import Graph
+
+
+def get_dependency_graph(application):
+    """
+    Get the dependencies of an application.
+
+    :param unicode application: The name of an application to get the
+        dependencies of.
+
+    :return tl.eggdeps.graph.Graph: Graph of Python dependencies, not
+        including ``application``.
+    """
+    dependency_graph = Graph()
+    dependency_graph.from_specifications([application])
+    dependency_graph.pop(application, None)
+    return dependency_graph
+
+
+def get_version():
+    """
+    :return str: The contents of the "VERSION" environment variable.
+
+    :raises Exception: If $VERSION is not set.
+    """
+    version = environ.get("VERSION")
+    if version is None:
+        raise Exception("Set the VERSION environment variable.")
+    return version
+
+
+def get_checksum(url):
+    """
+    Given the URL of a file, download that file and return its sha1 hash.
+
+    :param unicode url: A URL of a file.
+
+    :return str checksum: The sha1 hash of the file at ``url``.
+    """
+    try:
+        download = urlopen(url)
+    except HTTPError:
+        raise Exception("No file available at " + url)
+
+    checksum = sha1(download.read()).hexdigest()
+    download.close()
+    return checksum
+
+
+def get_class_name(version):
+    """
+    The ruby class name depends on the Flocker version. For example for version
+    0.3.0dev1 the class name should be Flocker0.3.0dev1.
+
+    :param str version: The version of Flocker this recipe is for.
+
+    :return str: The name of the ruby class needed if the file being created
+        is called "flocker-$VERSION.rb".
+    """
+    class_name = 'Flocker' + version
+    for disallowed_character in ['-', '.']:
+        class_name = class_name.replace(disallowed_character, '')
+    return class_name
+
+
+def get_formatted_dependency_list(dependency_graph):
+    """
+    :param tl.eggdeps.graph.Graph dependency_graph: Graph of Python
+        dependencies.
+
+    :return unicode: Space separated list of dependency names.
+    """
+    dependencies = []
+    for name, node in sorted(dependency_graph.iteritems()):
```

Comment:
I think you're intending to sort the returned  names alphabetically, so I'd say iterate of the `keys` alphabetically or better still do the sorting at the end when you join the list into  a string.